### PR TITLE
adding support to host multi-model endpoints

### PR DESCRIPTION
**Pull request by kmarquardsen**
### What does this PR do / how does this improve the operators?
adding support for https://docs.aws.amazon.com/sagemaker/latest/dg/multi-model-endpoints.html

### Which issue(s) does this PR fix?
N/A

Fixes # N/A

### Special notes for the reviewer:
in progress. need to add some tests

### Does this PR require changes to documentation?
will add example to samples

### All Submissions:

* [ ] Have you followed the guidelines in our Contributing document?
* [ ] Have you written or refactored unit tests to cover the change?
* [ ] Have you ran all unit tests and ensured they are passing?
* [ ] Have you manually tested each feature that is being added/modified?
* [ ] Have you ensured you have not introduced linting errors?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by akartsky — APPROVED** [review]

File: tests/codebuild/testfiles/xgboost-model.yaml [hunk 1]
```diff
@@ -5,8 +5,9 @@ metadata:
 spec:
   primaryContainer:
     containerHostname: xgboost
-    modelDataUrl: s3://{DATA_BUCKET}/batch-transform/xgboost-mnist/model.tar.gz
```
Comment (akartsky) [inline tests/codebuild/testfiles/xgboost-model.yaml:8]: can you please undo all changes to this file

I have included the `mode: MultiModel` in `tests/codebuild/testfiles/xgboost-hosting-deployment.yaml`

File: controllers/model/model_controller.go [hunk 2]
```diff
@@ -207,6 +210,13 @@ func (r *ModelReconciler) initializeContext(ctx *reconcileRequestContext) error
 	ctx.SageMakerClient = clientwrapper.NewSageMakerClientWrapper(r.createSageMakerClient(awsConfig))
 	ctx.Log.Info("Loaded AWS config")
 
+	if ctx.Model.Spec.PrimaryContainer.Mode == nil {
+		ctx.Model.Spec.PrimaryContainer.Mode = aws.String(DefaultContainerDefinitionMode)
+	}
+	for _, container := range ctx.Model.Spec.Containers {
+		container.Mode = aws.String(DefaultContainerDefinitionMode)
```
Comment (surajkota) [inline controllers/model/model_controller.go:217]: @akartsky check if its nil before setting value ?

what is special in aws.String?

Comment (akartsky) [reply controllers/model/model_controller.go:217]: oh wow 
nice catch

Comment (akartsky) [reply controllers/model/model_controller.go:217]: oh wow.. nice catch 

-----------------------

aws.String is a helper function to assign strings 
https://docs.aws.amazon.com/sdk-for-go/api/aws/

**If you don't use it we get error message**

`container.Mode = &DefaultContainerDefinitionMode` : cannot take address of DefaultContainerDefinitionMode (untyped string constant "SingleModel")
`container.Mode = DefaultContainerDefinitionMode` : cannot convert DefaultContainerDefinitionMode (untyped string constant "SingleModel") to *string

Comment (surajkota) [reply controllers/model/model_controller.go]: for sake of consistency can we use this - https://github.com/aws/amazon-sagemaker-operator-for-k8s/blob/master/controllers/controllertest/util.go#L32-L34

**Review by surajkota — APPROVED** [review]

**Pull request merged**